File: labs-opensource/M04-structured-output/solution/schema_and_data.py
"""
M04 Lab: Schema and Test Data (shared helper — identical to starter version)
==============================================================================
"""

import json
from typing import Optional

from pydantic import BaseModel


class ContactInfo(BaseModel):
    name: str
    email: str
    phone: Optional[str] = None
    company: Optional[str] = None
    role: Optional[str] = None


TEST_SIGNATURES = [
    "Best, Jane Smith | jane@acme.com | Acme Corp",
    "John Doe, Senior Engineer at MegaTech\njohn.doe@megatech.io | (555) 234-5678",
    "Cheers,\nDr. Maria García-López, Head of Research\nBioGen International\nmgarcia@biogen.int",
    "— Alex K. | Product @ StartupXYZ | alex@startupxyz.co | they/them",
    "Thanks!\nRobert \"Bob\" Williams III\nChief Financial Officer\nGlobal Finance Partners LLC\nrwilliams@gfp.com\n+1 (212) 555-0199",
]


if __name__ == "__main__":
    print(json.dumps(ContactInfo.model_json_schema(), indent=2))
    print(f"\nTest signatures: {len(TEST_SIGNATURES)}")

File: labs-opensource/M04-structured-output/solution/schema_and_data.js
/**
 * M04 Lab: Schema and Test Data (shared helper — identical to starter version)
 */

import { z } from "zod";

export const ContactInfo = z.object({
  name: z.string(),
  email: z.string().email(),
  phone: z.string().optional(),
  company: z.string().optional(),
  role: z.string().optional(),
});

export const CONTACT_PARAMETERS = {
  type: "object",
  properties: {
    name: { type: "string", description: "Full name of the person" },
    email: { type: "string", description: "Email address" },
    phone: { type: "string", description: "Phone number, if mentioned" },
    company: { type: "string", description: "Company name, if mentioned" },
    role: { type: "string", description: "Job title or role, if mentioned" },
  },
  required: ["name", "email"],
};

export const TEST_SIGNATURES = [
  "Best, Jane Smith | jane@acme.com | Acme Corp",
  "John Doe, Senior Engineer at MegaTech\njohn.doe@megatech.io | (555) 234-5678",
  "Cheers,\nDr. Maria García-López, Head of Research\nBioGen International\nmgarcia@biogen.int",
  "— Alex K. | Product @ StartupXYZ | alex@startupxyz.co | they/them",
  'Thanks!\nRobert "Bob" Williams III\nChief Financial Officer\nGlobal Finance Partners LLC\nrwilliams@gfp.com\n+1 (212) 555-0199',
];
